src/User32.Desktop/User32+DESKTOP_ACCESS_MASK.cs

```diff
@@ -0,0 +1,271 @@
+    public partial class User32
+    {
+        /// <summary>
+        /// The <see cref="Kernel32.ACCESS_MASK"/> type is a bitmask that specifies a set of access rights in the access mask of an access control entry.
+        /// This is a set of Specific Rights bitmask that defines access rights and security for desktop objects
+        /// </summary>
+        /// <remarks>
+        /// <see cref="Kernel32.ACCESS_MASK"/> is quite well described here: http://blogs.msdn.com/b/openspecification/archive/2010/04/01/about-the-access-mask-structure.aspx
+        /// Desktop security and access rights are quite well described here: https://msdn.microsoft.com/en-us/library/windows/desktop/ms687391(v=vs.85).aspx
+        /// </remarks>
+        public struct DESKTOP_ACCESS_MASK : IComparable, IFormattable, IComparable<DESKTOP_ACCESS_MASK>, IEquatable<DESKTOP_ACCESS_MASK>, IComparable<Kernel32.ACCESS_MASK>, IEquatable<Kernel32.ACCESS_MASK>
+        {
+            /// <summary>
+            /// Bits 28-31
+            /// </summary>
+            private const uint GenericRightsMask = 0xf0000000;
+
+            /// <summary>
+            /// Bits 24-27
+            /// </summary>
+            private const uint SpecialRightsMask = 0x0f000000;
+
+            /// <summary>
+            /// Bits 16-23
+            /// </summary>
+            private const uint StandardRightsMask = 0x00ff0000;
+
+            /// <summary>
+            /// Bits 0-15
+            /// </summary>
+            private const uint SpecificRightsMask = 0x0000ffff;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="DESKTOP_ACCESS_MASK"/> struct.
+            /// </summary>
+            /// <param name="value">The value for the <see cref="DESKTOP_ACCESS_MASK"/>.</param>
+            public DESKTOP_ACCESS_MASK(uint value)
+            {
+                this.Value = value;
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="DESKTOP_ACCESS_MASK"/> struct.
+            /// </summary>
+            /// <param name="value">The reference <see cref="Kernel32.ACCESS_MASK"/> for the <see cref="DESKTOP_ACCESS_MASK"/>.</param>
+            public DESKTOP_ACCESS_MASK(Kernel32.ACCESS_MASK value)
+            {
+                this.Value = value;
+            }
+
+            /// <summary>
+            /// The following are the generic access rights for a desktop object contained in the interactive window station of the user's logon session
+            /// </summary>
+            [Flags]
+            public enum GenericRight : uint
+            {
+                GENERIC_READ = Kernel32.ACCESS_MASK.StandardRight.STANDARD_RIGHTS_READ |
+                    SpecificRight.DESKTOP_ENUMERATE |
+                    SpecificRight.DESKTOP_READOBJECTS,
+
+                GENERIC_WRITE = Kernel32.ACCESS_MASK.StandardRight.STANDARD_RIGHTS_WRITE |
+                    SpecificRight.DESKTOP_CREATEMENU |
```

Comment:
On this page, we have an update that reflects the semantics just for desktop objects:

[MSDN Desktop Security and Access Rights](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682575(v=vs.85).aspx),

It states:

> The following are the generic access rights for a desktop object contained in the interactive window station of the user's logon session.
> ```
> |-----------------+----------------------|
> | Access right    | Description          |
> |-----------------+----------------------|
> | GENERIC_READ    | DESKTOP_ENUMERATE    |
> |                 | DESKTOP_READOBJECTS  |
> |                 | STANDARD_RIGHTS_READ |
> |-----------------+----------------------|
> ```

and so on...

We have the same semantic mapping for Window Stations here: [MSDN Window Station Security and Access Rights](https://msdn.microsoft.com/en-us/library/windows/desktop/ms687391(v=vs.85).aspx)

I really tried to map everything on the Desktop and Window Station APIs, including types, functions and the security/access mappings because I am using them internally.